### PR TITLE
[JIMT] more video analytics events

### PR DESCRIPTION
**Pull request by thomasoniii**
Youtube actually has two separate domains - [youtube.com](http://youtube.com/) and [youtube-nocookie.com](http://youtube-nocookie.com/). It’s significant because we load our videos from [youtube-nocookie.com](http://youtube-nocookie.com/) , but the youtube api is loaded from [youtube.com](http://youtube.com/).

Okay, So there are several combinations that can happen -
- [youtube.com](http://youtube.com/) and [youtube-nocookie.com](http://youtube-nocookie.com/) are not blocked by the school - we load the video from youtube, and we get all of our analytics events.
- [youtube.com](http://youtube.com/) is blocked, but [youtube-nocookie.com](http://youtube-nocookie.com/) is not blocked. We load the video from youtube, but we do not get analytics events. This one’s the problem.
- Both domains are blocked. We load the video locally from s3, and do not currently get any analytics events. however, I have a PR ready that’ll track all the same events for our locally housed videos, so this isn’t a big idea.

The first case was addressed in #54648. 

This PR addresses the third case where both domains are blocked and we fall back to our own video and our own video player.

Detailing the middle case, which is _not_ in this PR, but should be circled back to if possible - What ends up happening is that we can successfully load the video from youtube, but we cannot load the youtube api (since it’s only vended out from a [youtube.com](http://youtube.com/) domain). This means that we can’t attach the amplitude event emitters to the video, so we can track nothing.

The way to fix this would be to load up the youtube iframe_api from a different location…but I don’t have one of those yet. Unfortunately, they don’t seem to vend it from the [youtube-nocookie.com](http://youtube-nocookie.com/) domain, and I’ve been reading mixed recommendations about whether we can locally copy down the file and vend it out ourselves. There are a host of potential problems if we do that since we’d be operating outside of an official youtube release, so we should tread carefully on that route. I just don’t know enough about it yet to make a recommendation.

Anyway, we're gonna circle back to that case. We can also potentially tack on more events to track which case the user is in explicitly. But once again, we'll circle back to that once it's better established.

## Warning!!

We have entered Pixel Lock for Hour of Code!

Computer Science Education Week will be happening from Dec 4 - Dec 10. Alongside this event, we will
be launching our new Hour of Code activity. Please consider any risk introduced in this PR that
could affect Dance Lab, instructions, saving and logging student progress, caching, or anything
related to the Hour of Code activities, old or new. Even small changes, such as a different button
color, are considered significant during this time. If this change will affect the new Hour of Code
activity in any way, join the morning change review to get your changes approved prior to merging.
Reach out to the Student Labs team for more details!

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/videos.js [hunk 1]
```diff
@@ -515,7 +516,26 @@ function addFallbackVideoPlayer(videoInfo, playerWidth, playerHeight) {
     }
   );
 
-  videoPlayer.on('ended', onVideoEnded);
+  const analyticsData = {
+    url: location.href,
+    video: videoInfo.download,
+    fallback: 'code.org',
```
Comment (sanchitmalhotra126) [inline apps/src/code-studio/videos.js:522]: Is `fallback: 'code.org'` our indication that we used the fallback player? Are there other values `fallback` could have?

**Review by sanchitmalhotra126 — APPROVED** [review]
One question, but code LGTM!

Re the "middle" case - do we at least have analytics on when this case happens (video loaded but can't listen to API events)?

**Comment by thomasoniii** [comment]
> One question, but code LGTM!
> 
> Re the "middle" case - do we at least have analytics on when this case happens (video loaded but can't listen to API events)?

We do not. 😞 

That was my last line in the summary - that we could potentially track what state the user is in (youtube+youtube-nocookie, just youtube-nocookie, or neither of them) and fire off an event stating it, but noted as a follow up item. We definitely should add something like this, though.

File: apps/src/code-studio/videos.js (hunk 1 above)
Comment (thomasoniii) [reply apps/src/code-studio/videos.js:522]: Correct, just an explicit note that it's using our locally hosted video.

Presumably we'd also have a `fallback : youtube-blocked` or the equivalent for the middle case, once we're able to track those events on it. Or, alternatively, we may just decide that just knowing the video url is sufficient. Def something that we can iterate on.

**Pull request merged**